pkg/inclusion/nmt_caching.go

```diff
@@ -0,0 +1,125 @@
+package inclusion
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/rsmt2d"
+	"github.com/tendermint/tendermint/pkg/da"
+	"github.com/tendermint/tendermint/pkg/wrapper"
+)
+
+// TODO optimize https://github.com/celestiaorg/nmt/blob/e679661c6776d8a694f4f7c423c2e2eccb6c5aaa/subrootpaths.go#L15-L28
+
+// subTreeRootCacher keep track of all the inner nodes of an nmt using a simple
+// map. Note: this cacher does not cache individual leaves or their hashes, only
+// inner nodes.
+type subTreeRootCacher struct {
+	cache map[string][2]string
+}
+
+func newSubTreeRootCacher() *subTreeRootCacher {
+	return &subTreeRootCacher{cache: make(map[string][2]string)}
+}
+
+// Visit fullfills the nmt.VisitorNode function definition. It stores each inner
```

Comment:
Ref: https://github.com/celestiaorg/nmt/blob/master/nmt.go#L22
```suggestion
// Visit fullfills the nmt.NodeVisitorFn function definition. It stores each inner
```